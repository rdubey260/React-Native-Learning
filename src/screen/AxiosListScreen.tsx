import { useEffect, useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import axios from 'axios';


interface Iitem {
  id: string;
  title: string;
  thumbnailUrl: string;
}

export const RenderItem = ({ item, handlePress }) => {
  return (
    <TouchableOpacity testID="item-press" onPress={() => handlePress(item)}>
      <View style={{ marginVertical: 10, alignItems: 'center' }}>
        <Image src={item.thumbnail} style={{ height: 200, width: '90%' }} />
        <Text>{item.brand}</Text>
      </View>
    </TouchableOpacity>
  );
};

const AxiosListScreen = () => {
  const [data, setData] = useState([])

  useEffect(() => {
    axios.get('https://dummyjson.com/products').then((res) => {
      console.log('>>>>>>>res', res.data)
      setData(res.data.products)
    }).catch((err) => {
      console.log('>>>>>>>err', err)
    })
  }, [])

  const handlePress = (item: Iitem) => {
    console.log('>>>>>');
  };

  return (
    <View testID="list-container" style={styles.container}>
      <FlatList
        testID="flat-list"
        data={data}
        renderItem={({ item }) => (
          <RenderItem item={item} handlePress={handlePress} />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    flex: 1,
  },
});

export default AxiosListScreen;
