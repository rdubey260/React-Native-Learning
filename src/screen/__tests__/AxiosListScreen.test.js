import React from 'react';
import { render } from '@testing-library/react-native';
import axios from "axios";
import AxiosListScreen from '../AxiosListScreen';

jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve()),
}));

const url = "https://dummyjson.com/products";

test('render the settings UI', () => {
  const setting = render(<AxiosListScreen />);
  const tree = setting.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should return todo list', async () => {
  //Our desired output
  const data1 = {
    limit: 30,
    products: [{
      brand: "Apple",
      category: "smartphones",
      description: "An apple mobile which is nothing like apple",
      discountPercentage: 12.96,
      id: 1,
      images: [
        ""
      ],
      price: 549,
      rating: 4.69,
      stock: 94,
      thumbnail: "https://cdn.dummyjson.com/product-images/1/thumbnail.jpg",
      title: "iPhone 9"
    }],
    skip: 0,
    total: 100
  }

  //Prepare the response we want to get from axios
  const mAxiosResponse = {
    data: data1,
    config: {},
    headers: {},
    request: {},
    status: 200,
    statusText: undefined
  };
  jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);
  render(<AxiosListScreen />);
});


