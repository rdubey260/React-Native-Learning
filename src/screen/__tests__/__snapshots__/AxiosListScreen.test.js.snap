// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the settings UI 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
  testID="list-container"
>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="flat-list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
